website/static/js/nodesDelete.js

```diff
@@ -280,24 +282,105 @@ NodesDeleteViewModel.prototype.back = function() {
     var self = this;
+    self.confirmationString = $osf.getConfirmationString();
+    self.QUICKDELETE = 'quickDelete';
+    self.SELECT = 'select';
+    self.CONFIRM = 'confirm';
+    self.page = ko.observable(self.QUICKDELETE);
+    self.confirmInput = ko.observable('');
+    self.canDelete = ko.observable(false);
+    self.nodeType = nodeType;
+    self.nodeApiUrl = nodeApiUrl;
+    self.isPreprint = isPreprint;
+    self.message = ko.computed(function(){
+        var preprintMessage = '<br><br>This ' + self.nodeType + ' contains a <strong>preprint</strong>. Deleting this ' +
+          self.nodeType + ' will also delete your <strong>preprint</strong>. This action is irreversible.';
+        var message = 'It will no longer be available to other contributors on the project.' +
+        (self.isPreprint ? preprintMessage : '');
+
+        return message;
+    });
+    self.atMaxLength = ko.observable(false);
+    self.nodesDeleted = ko.observable(true);
+    self.pageTitle = ko.computed(function() {
+        return 'Are you sure you want to delete this ' + self.nodeType + '?';
+    });
+};
 
-    self.selector = selector;
-    self.$element = $(self.selector);
-    self.viewModel = new NodesDeleteViewModel(node);
-    self.viewModel.fetchNodeTree().done(function(response) {
-        new NodesDeleteTreebeard('nodesDeleteTreebeard', response, self.viewModel.nodesState, self.viewModel.nodesOriginal);
+QuickDeleteViewModel.prototype.clear = function(){
+    var self = this;
+    self.nodesDeleted = ko.observable(false);
+};
+
+QuickDeleteViewModel.prototype.handleEditableUpdate = function(element) {
+    var self = this;
+    var $element = $(element);
+    var inputText = $element[0].innerText;
+    var inputTextLength = inputText.length;
+
+    self.atMaxLength(inputTextLength >= self.confirmationString.length);
+    self.canDelete(inputText === self.confirmationString);
+};
+
+QuickDeleteViewModel.prototype.confirmChanges = function(){
+    var self = this;
+    var request = $.ajax({
+        type: 'DELETE',
+        dataType: 'json',
+        url: self.nodeApiUrl
     });
-    self.init();
+    request.done(function(response) {
+        // Redirect to either the parent project or the dashboard
+        window.location.href = response.url;
+    });
+    request.fail($osf.handleJSONError);
+};
+
+function NodesDelete(childExists, nodeType, isPreprint, nodeApiUrl) {
+    var self = this;
+
+    self.selector = '#nodesDelete';
+    self.$element = $(self.selector);
+
+    if (childExists) {
+        self.viewModel = new NodesDeleteViewModel(nodeType, isPreprint, nodeApiUrl);
+        self.viewModel.fetchNodeTree().done(function(response) {
+            new NodesDeleteTreebeard('nodesDeleteTreebeard', response, self.viewModel.nodesState, self.viewModel.nodesOriginal);
+        });
+    } else {
+        self.viewModel = new QuickDeleteViewModel(nodeType, isPreprint, nodeApiUrl);
+    }
```

Comment:
Logic and naming of view models is very readable here: if a project has components, use the NodesDeleteViewModel, otherwise QuickDeleteViewModel. 👍